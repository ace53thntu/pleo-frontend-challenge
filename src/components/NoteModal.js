import React from "react";
import Modal from "react-modal";

Modal.setAppElement("#app");

export default ({ isOpen, handleAddNote, toggleModal }) => (
    <Modal isOpen={isOpen} contentLabel="Example Modal">
        <div>Add Note</div>
        <div>Under 150 characters</div>
        <form onSubmit={handleAddNote}>
            <textarea
                name="comment-text"
                placeholder="Add a note about this purchase"
                maxLength="150"
                className="text-area"
            />
            <button className="button">Confirm</button>
            <button
                className="button--secondary"
                type="button"
                onClick={toggleModal}
            >
                Cancel
            </button>
        </form>
    </Modal>
);

//Remember to take the modal fix out of the base style settings and move it into the model component setting file
