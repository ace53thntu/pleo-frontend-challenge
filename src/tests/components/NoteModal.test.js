import React from "react";
import { shallow } from "enzyme";
import NoteModal from "../../components/NoteModal";

it("should render note modal", () => {
    const wrapper = shallow(<NoteModal isOpen={true} />);
    expect(wrapper).toMatchSnapshot();
});
