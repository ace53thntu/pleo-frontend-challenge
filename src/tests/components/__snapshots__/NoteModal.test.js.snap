// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render note modal 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  closeTimeoutMS={0}
  contentLabel="Note Modal"
  isOpen={true}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div>
    Add Note
  </div>
  <div>
    Under 150 characters
  </div>
  <form>
    <textarea
      className="text-area"
      maxLength="150"
      name="comment-text"
      placeholder="Add a note about this purchase"
    />
    <button
      className="button"
    >
      Confirm
    </button>
    <button
      className="button--secondary"
      type="button"
    >
      Cancel
    </button>
  </form>
</Modal>
`;
